### ISSUE TITLE
Windows: End point disconnect event doesn't occur in Windows

### ISSUE THREAD

**Issue by NCrashed** [haskell-distributed/network-transport-tcp#35]
Windows 7 x64
Haskell platform 2014.2.0.0
network-transport-tcp-0.4.1

Transport events `EndPointClosed` and `ConnectionClosed` are never raised. Channel is considered alive even the other side is dead.

**Comment by NCrashed** [comment]
On server side (executed in Windows) at moment of disconnection appears following message:
```
ae-server.exe: mallocPlainForeignPtrBytes: size must be >= 0
```

With server executed in Fedora 21 windows client also doesn't close connection at exit but server doesn't spam the message.

**Comment by qnikst** [comment]
Looks very strange, at least `EndPointClosed` is supposed to be sent when endpoint is going down.

Do you have some sort of minimal example? or at least try to run tests for the package, this could help to understand where to start looking, meanwhile I'll try to find a windows host to debug that.

**Comment by NCrashed** [comment]
Fixed. The problem was in my legacy code. Library didn't provide `ntohl` function for Windows platform approx. two years ago and I included custom version. And now the old `ntohl` overrides library one and produces the bug. No legacy ntohl - no problem. (Fixed by #30).
